[Tooling] Remove release_notes.xml file now that changelog upload has been automated via supply

AliSoftware opened the pull request:
Back in https://github.com/wordpress-mobile/WordPress-Android/pull/16235, I implemented the automation of uploading the `changelogs/*.txt` release notes to the PlayStore using fastlane's `upload_to_play_store` (aka `supply`)

While this seemed to have worked great, so far I kept the `android_create_xml_release_notes` action around a little longer just in case I would notice issues with the new changelog upload automation.

This PR prepares the removal of the remaining traces of `android_create_xml_release_notes` (and the `release_notes.xml` files it created) so that we can land it when we're ready.

ℹ️  I'm marking it with the `Not Ready for Merge` label only because we might want to wait to see if the submission of 19.6 worked before merging. (Though honestly if we prefer to land it before, we could always revert in case some errors with the new automation start to appear and we need the XML it back until we fix them…)

## To test

Nothing to test on the PR per se. But cc-ing @oguzkocer as you'll cover for me for the `finalize_release` of 19.6 and the code freeze of 19.7:
 - Once you you'll run `finalize_release` for 19.6 then go to the PlayStore to submit it to Google for review, don't hesitate to double-check that the automation properly updated the release notes for all locales in the PlayStore, with the correct copies for 19.6, and without you having to paste them manually (nor fix any issue manually on those notes)
 - Then if that worked correctly, feel free to remove the `[Status] Not Ready for Merge` label, then merge this PR in `trunk` before doing the code-freeze for `19.7` (or retarget this PR to `release/19.7` after the code freeze otherwise).

AliSoftware committed:
Remove calls to `android_create_xml_release_notes`

Not needed anymore now that the upload of changelogs/*.txt have recently been automated via `upload_to_play_store`

See https://github.com/wordpress-mobile/WordPress-Android/pull/16235

AliSoftware committed:
Remove old `release_notes.xml` files

Now that they won't be needed nor generated anymore

AliSoftware commented on fastlane/lanes/localization.rb:
```diff
@@ -180,7 +177,6 @@
       skip_upload_aab: true,
       skip_upload_metadata: false,
-      # TODO: enable changelogs upload once we drop the `android_create_xml_release_notes` action call (and drop the step in release scenario)
       skip_upload_changelogs: true,
```
Removed that TODO without actually enabling `skip_upload_changelogs: false` here like it initially suggesteed after all, because I realized that this TODO that I wrote back then… was not correct.

Indeed, this `upload_playstore_localized_metadata ` lane is still intended to only update the metadata (as in: app title, short description, long description, …), and for production builds only (hence the `track: 'production'` on L177), and not related to the changelogs (which are uploaded during a different lane, namely `upload_build_to_play_store`, at the same time we upload the `apk`/`aab` after a build)

AliSoftware commented on fastlane/lanes/localization.rb:
```diff
@@ -169,9 +169,6 @@
     metadata_dir = File.join('fastlane', APP_SPECIFIC_VALUES[app.to_sym][:metadata_dir], 'android')
     version_code = android_get_release_version['code']
 
-    # TODO: This is currently a separate lane so we can invoke it separately, but if after a couple of sprints we see that it works well,
-    #       it we would make sense to remove the lane and instead directly use `skip_upload_metadata: (options[:track] != 'production')`
-    #       in the `upload_build_to_play_store` lane? (especially if we do the same for `skip_upload_changelogs` and get rid of `android_create_xml_release_notes`)
```
This `skip_upload_metadata: (options[:track] != 'production'` idea has already been applied and taken care of by https://github.com/wordpress-mobile/WordPress-Android/pull/16235/files#r840866181 which is why this TODO is ok to remove now.

I've decided to still keep that lane around, even if the automation of the upload of both the metadata (app title, description…) and changelogs (aka release notes) have been automated during `upload_build_to_play_store` by https://github.com/wordpress-mobile/WordPress-Android/pull/16235/files#r840866181, because I figured it could still be useful to trigger an update of the metadata manually / on demand and decoupled from the upload of a new build. Especially since Google allows for the metadata (Play Store Listing) to be updated separately from submitting a new version of the app if we want to.

mokagio commented on fastlane/lanes/localization.rb:
```diff
@@ -158,7 +158,7 @@
   end
 
-  # Updates the metadata (from `fastlane/{metadata|jetpack_metadata}/android/*/*.txt`) in the PlayStore (Main store listing).
+  # Updates the metadata in the PlayStore (Main store listing) from the content of `fastlane/{metadata|jetpack_metadata}/android/*/*.txt` files
```
Super nitpick:

```suggestion
  # Updates the metadata in the Play Store (Main store listing) from the content of `fastlane/{metadata|jetpack_metadata}/android/*/*.txt` files
```

Source:

<img width="599" alt="image" src="https://user-images.githubusercontent.com/1218433/162669277-674ab48e-10ba-46c6-8f51-01140ead03a3.png">

mokagio commented on fastlane/lanes/localization.rb:
```diff
@@ -229,9 +225,9 @@
-    android_create_xml_release_notes(download_path: download_path, build_number: "#{options[:build_number]}", locales: RELEASE_NOTES_LOCALES)
-    sh("git add #{download_path} && git commit -m \"Update WordPress metadata translations for #{options[:version]}\" && git push origin HEAD")
+    git_add(path: download_path)
+    git_commit(message: "Update WordPress metadata translations for #{options[:version]}", allow_nothing_to_commit: true)
```
```suggestion
    git_commit(path: download_path, message: "Update WordPress metadata translations for #{options[:version]}", allow_nothing_to_commit: true)
```

I tried this locally and it asked me for input, which I don't think it's what we want here.

<img width="1497" alt="Screen Shot 2022-04-11 at 3 32 09 pm" src="https://user-images.githubusercontent.com/1218433/162671300-ff16b311-9136-4735-bcfc-6f3674c79d4c.png">

mokagio replied:
I wanted to make sure my suggestion was correct so I tested it locally, might as well open a PR for it while at it: https://github.com/wordpress-mobile/WordPress-Android/pull/16290.

AliSoftware replied:
Addressed by merging your PR #16290 into this branch 🙂 🙇

mokagio commented on fastlane/lanes/localization.rb:
```diff
@@ -264,10 +260,9 @@
-    android_create_xml_release_notes(download_path: download_path, build_number: "#{options[:build_number]}", locales: locales_including_enUS)
-    sh("git add #{download_path} && git commit -m \"Update Jetpack metadata translations for #{options[:version]}\" && git push origin HEAD")
+    git_add(path: download_path)
+    git_commit(message: "Update Jetpack metadata translations for #{options[:version]}", allow_nothing_to_commit: true)
```
See comment above

```suggestion
    git_commit(path: download_path, message: "Update Jetpack metadata translations for #{options[:version]}", allow_nothing_to_commit: true)
```

mokagio replied:
I wanted to make sure my suggestion was correct so I tested it locally, might as well open a PR for it while at it: https://github.com/wordpress-mobile/WordPress-Android/pull/16290.

AliSoftware replied:
Addressed by merging your PR #16290 into this branch via a1af416 🙂 🙇

mokagio committed:
Add `path` parameter to `git_commit` call when downloading metadata

mokagio reviewed (changes requested):
See my comments about the missing `path` in the `git_commit` calls. I'm requesting changes because they break the automated flow possibly resulting in a lot of time where the lane remains waiting for input and it's best to address that sooner rather than later

oguzkocer commented:
@AliSoftware @mokagio I think it makes sense to wait to merge this PR until we verify that the release notes are uploaded correctly - as described by @AliSoftware in the PR description. I'll merge this to `trunk` before `code_freeze` if I don't forget about it 😅 (and of course if it's approved by then)

AliSoftware committed:
Fix typo in code comment

Co-authored-by: Gio Lodi <giovanni.lodi42@gmail.com>

AliSoftware commented:
@mokagio I've addressed your feedback — by merging your PR, and applying your typo fix suggestion. Thanks!

@oguzkocer I've manually added a step in the Release Scenario of 19.6 to mention merging this PR after submitting WP and JP to PlayStore 😉

oguzkocer commented:
> @oguzkocer I've manually added a step in the Release Scenario of 19.6 to mention merging this PR after submitting WP and JP to PlayStore 😉

Thank you for doing that! 🙇‍♂️

oguzkocer commented:
@AliSoftware The release notes were successfully uploaded to Play Store. There is a step in the release scenario to merge this PR in, but since it's not approved yet, I am going to skip it.

I see that you've addressed Gio's feedback, but I don't want to override his review especially since it's not urgent to merge this PR as we can re-target it to the next release.

@mokagio Can you please follow up with your review when you get a chance?

AliSoftware commented:
@mokagio I've rebased this PR on top of `release/19.7`, fixed conflicts along the way, and updated the PR's base branch accordingly. So the only remaining thing to get this to land is your PR approval now 🙂 🙏

mokagio reviewed (approved):
:shipit: 

Sorry folks for missing this. 

As a note for future occurrences, since my only comment was about the bug and the PR I submitted fixed it, IMHO it would have been acceptable to override my rejection and merge, given I was unresponsive.
